allennlp/common/file_utils.py

```diff
@@ -57,7 +70,17 @@ def get_from_cache(url: str, cache_dir: str = DATASET_CACHE) -> str:
     If it's not there, download it. Then return the path to the cached file.
     """
     os.makedirs(cache_dir, exist_ok=True)
-    path = os.path.join(cache_dir, url_to_filename(url))
+
+    # make HEAD request to check ETag
+    response = requests.head(url)
+    if response.status_code != 200:
+        raise IOError("HEAD request failed for url {}".format(url))
```

Comment:
What if you don't have an internet connection and want to run something on your laptop?

Reply:
if your experiment uses actual filenames, then nothing, this code path shouldn't even be run. 

if your experiment uses datasets-in-the-cloud cached locally, then yes this would break your workflow. is that an important use case?

Reply:
Probably not, and it's easy to fix later.